sql/core/src/main/scala/org/apache/spark/sql/execution/SortPrefixUtils.scala

```diff
@@ -66,6 +66,29 @@ object SortPrefixUtils {
   }
 
   /**
+   * Returns whether the specified SortOrder can be satisfied with a radix sort on the prefix.
+   */
+  def canSortFullyWithPrefix(sortOrder: SortOrder): Boolean = {
+    sortOrder.dataType match {
+      case BooleanType | ByteType | ShortType | IntegerType | LongType | DateType | TimestampType =>
```

Comment:
Because prefix is not nullable, so one minimum value is chosen to represent `null`. For LongType,  by only compare the prefix, it can't tell the value is `null` or the minimum value (Long.MinValue). This may also apply to DecimalType and DoubleType